src/com/sap/piper/Utils.groovy

```diff
@@ -61,6 +61,39 @@ def stashWithMessage(name, msg, include = '**/*.*', exclude = '', useDefaultExcl
     }
 }
 
+def stashStageFiles(Script script, String stageName) {
+    try {
+        List stashes = script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.stashes ?: []
+
+        stashList(script, stashes)
+
+        //NOTE: We do not delete the directory in case Jenkins runs on Kubernetes.
+        // deleteDir() is not required in pods, but would be nice to have the same behaviour and leave a clean fileSystem.
+        if (!isInsidePod(script)) {
+            script.deleteDir()
+        }
+    } catch (e) {
+        echo "Failed to stash files for stage $stageName (${e.getMessage()})"
```

Comment:
IMO we should be careful with swallowing exceptions. I'm not sure if writing it into the log and continue is appropriate here. What will happen later down the road in case a stash is missing? I guess some file cannot be materialized somewhere. In the best case we get an Exception later. In the worst case the pipeline succeeds silently.

Wouldn't it be better to be failfast here and let the Exception bubble up?

Reply:
I thought I would be imitating the code which already exists in other stashing related Utils methods. I will just remove the try-block entirely.